packages/core/test/discovery.test.js

```diff
@@ -1970,4 +1970,68 @@ describe('Discovery', () => {
       });
     });
   });
+
+  describe('Service Worker =>', () => {
+    it('captures original request', async () => {
+      server.reply('/sw.js', () => [200, 'text/javascript', dedent`
+      const fetchUpstream = async(request) => {
+        return await fetch(request.clone());
+      }
+      
+      self.addEventListener('fetch', (event) => {
+        const { request } = event
+        event.respondWith(fetchUpstream(request));
+      });
+      
+      self.addEventListener("activate", (event) => {
+        event.waitUntil(clients.claim());
+      });
+      `]);
+
+      server.reply('/app.js', () => [200, 'text/javascript', dedent`
+      const registerServiceWorker = async () => {
+        await navigator.serviceWorker.register('sw.js',{ scope: './', });
+      };
+      
+      await registerServiceWorker();
```

Comment:
didnt know we can call await in base js context

Reply:
The script declaration is done as -
```html
    <script type="module" src="app.js"></script>
```

Here, `type="module"` allows for using `await` at top-level.
If we remove `type="module"` it throws error as -

`Uncaught SyntaxError: await is only valid in async functions and the top level bodies of modules (at app.js:5:1)`

Reply:
Oh nice, did not know this :P